tests/std/tests/P0896R4_ranges_alg_is_permutation/test.cpp

```diff
@@ -134,6 +134,17 @@ struct instantiator {
             assert(ranges::is_permutation(
                 r1, r2, {}, [](int n) { return n + 1; }, [](int n) { return n - 1; }));
         }
+        { // Test GH-2888: `<algorithm>`: `ranges::is_permutation`'s helper lambda does not specify return type
+            struct NonCopyableBool {
+                constexpr operator bool() {
+                    return true;
+                }
+
+                NonCopyableBool()                       = default;
+                NonCopyableBool(const NonCopyableBool&) = delete;
+            } b;

```

Comment:
No change requested: We usually avoid defining a struct and a variable together. Separating the definitions consumes an extra line, but makes it clearer that two things are being defined (similar rationale to avoiding `Type obj1, obj2;`).

This stylistic nitpick isn't worth resetting testing, but I wanted to note it for the future.